### ISSUE TITLE
Getting an `undefined symbol` error during installation (could not load parser for typst)

### ISSUE THREAD

**Issue by Andrew15-5** [uben0/tree-sitter-typst#3]
I've transitioned from an old init.vim to init.lua via [NvChad](https://nvchad.com). And I'm new to many things, including tree-sitter. I did a `:TSInstall typst` from which I got:

```
Installation not possible: ...vim/lazy/nvim-treesitter/lua/nvim-treesitter/install.lua:86: Parser not available for language "typst"                                                                                 
See https://github.com/nvim-treesitter/nvim-treesitter/#adding-parsers on how to add a new parser!                                                                                                                   
```

So then I went to https://github.com/nvim-treesitter/nvim-treesitter/#adding-parsers. From step 3 I copied and modified this:

```lua
local parser_config = require("nvim-treesitter.parsers").get_parser_configs()
parser_config.typst = {
  install_info = {
    url = "https://github.com/uben0/tree-sitter-typst.git",
    -- url = "/tmp/tree-sitter-typst", -- Both produce the same error.
    files = { "src/parser.c" }, -- note that some parsers also require src/scanner.c or src/scanner.cc
    -- optional entries:
    -- branch = "main", -- default branch in case of git repo if different from master
    -- generate_requires_npm = false, -- if stand-alone parser without npm dependencies
    requires_generate_from_grammar = true, -- if folder contains pre-generated src/parser.c
  },
}
```

I also installed a `tree-sitter-cli` from crates.io. But then running `:TSInstall typst` or `:TSInstallFromGrammar typst` results in the same error:

```
[nvim-treesitter] [0/1] Generating source files from grammar.js...                                                                                                                                                   
[nvim-treesitter] [0/1] Compiling...                                                                                                                                                                                 
Could not load parser for typst: "Failed to load parser for language 'typst': uv_dlopen: /home/andrew/.local/share/nvim/lazy/nvim-treesitter/parser/typst.so: undefined symbol: tree_sitter_typst_external_scanner_cr
eate"                                                                                                                                                                                                                
[nvim-treesitter] [1/1] Treesitter parser for typst has been installed                                                                                                                                               
```

I also tried cloning the repo locally and running `tree-sitter generate|test` which got me the same error.

At the beginning of all this, I stumbled upon this: https://github.com/tree-sitter/tree-sitter/discussions/2153 because in my old vim-plug setup I used onedark theme and typst.vim plugin for syntax highlighting. But now with nvchad I still get _some_ syntax highlighting, but it's not the same as, for example, `none` and `auto` are shown with the same color as most stuff (strings and functions are different): ![image](https://github.com/uben0/tree-sitter-typst/assets/37143421/760e5cd9-347c-48db-8565-3b64715d52f2). But when I do `:colorscheme onedark` it makes everything as it was with my previous setup: ![image](https://github.com/uben0/tree-sitter-typst/assets/37143421/e08721aa-1ca6-4b7f-8be2-b17a0c9a2432).

So I figured that nvchad uses tree-sitter for everything (because tree-sitter is configured in its base config) and that it's probably not the problem of nvchad, but instead of tree-sitter. More specifically, the "lack of Typst support". And here is when we get to this Typst tree-sitter project. Could you please help me with the error? Did I do something wrong?

Everything till this point was installed automatically after a "clean install" of Neovim. Now, I don't know if this non-default Typst tree-sitter can be also automatically installed (at least after the first generation and stuff like that). Although this is a different question.

**Comment by Ziqi-Yang** [comment]
It seems like the error is caused by the lack of `src/scanner.c` file. ( After removing this file, my `tree-sitter generate` produced the same error as yours.  I think you should change
```lua
files = { "src/parser.c" }, -- note that some parsers also require src/scanner.c or src/scanner.cc
```
to
```lua
files = { "src/parser.c", "src/scanner.c" }, -- note that some parsers also require src/scanner.c or src/scanner.cc
```
in your configuration.
> I also tried cloning the repo locally and running tree-sitter generate|test which got me the same error.

I still don't understand how your machine produces this error after a fresh clone of this repository.

**Comment by Andrew15-5** [comment]
Ok, so it seems that the problem has to do with what is opened. When there is an empty buffer, then installation is a breeze, but if I open a Typst file, then the error occur. When I added the scanner, then the issue was gone.

Ok, so now installation was a success. But despite that, the syntax highlighting isn't shown. When I added Markdown tree-sitter, then it was automatically applied and everything was obvious. With the Typst tree-sitter, after re-open everything is in one color (I added the filetype and LSP was running). Now I wonder why I don't have pretty colors (I disabled the kaarmu/typst.vim plugin).
